src/main/flight/pid.c

```diff
@@ -278,19 +278,24 @@ void pidResetIterm(void)
 
 void pidUpdateTpaFactor(float throttle)
 {
+    // always allow TPA low on arming
     static bool isTpaLowFaded = false;
     // don't permit throttle > 1 & throttle < 0 ? is this needed ? can throttle be > 1 or < 0 at this point
     throttle = constrainf(throttle, 0.0f, 1.0f);
-    bool isThrottlePastTpaLowBreakpoint = (throttle < pidRuntime.tpaLowBreakpoint && pidRuntime.tpaLowBreakpoint > 0.01f) ? false : true;
+
+    const bool tpaLowInactive = (pidRuntime.tpaLowDisabled || isTpaLowFaded || (throttle >= pidRuntime.tpaLowBreakpoint));
     float tpaRate = 0.0f;
-    if (isThrottlePastTpaLowBreakpoint || isTpaLowFaded) {
+    if (tpaLowInactive) {
+        // to get here, throttle exceeded the tpaLow threshold
+        // set the 'faded' latch when not configured to be always on
+        isTpaLowFaded = !pidRuntime.tpaLowAlways;
```

Comment:
since this is the only place where `isTpaLowFaded` can be changed, do we even need this static variable?

Reply:
I agree.  We need a static value so that if the user wants TPA Low only at the start, they get it until the throttle goes high, then it stops.  So currently `isTpaLowFaded` handles that latching, starting off false, and becoming true when the threshold is exceeded.  

I had though that perhaps we could write over the runtime value of `pidRuntime.tpaLowDisabled` when we reach that threshold, setting to `!pidRuntime.tpaLowAlways`.  If we didn't want it always, even if we enabled it at the start by setting this would block it.  If we disabled it at the start by setting a ridiculously low tpa breakpoint, the same would happen. I seems right, but I wasn't sure about it.

I'd need to test all this and as yet it is not tested.

Reply:
@limonspb  OK made a commit that implements your idea, LGTM, but needs testing.  
Thanks for the feedback.

Reply:
hey guys, i developed the original PR with matlab. let me know when this is ready and you think double checking the new implementation is worth it/beneficial

Reply:
Yes would be a good unit test to add, hasn't been done though.
@Quick-Flash - could you do it?  I cannot get my head around unit tests, they drive me nuts.

Reply:
@ctzsnooze or @limonspb could you point me to a "good" example of a unit test within the bf codebase? i might give it a shot.

Reply:
@ledvinap  ?

Reply:
> Yes would be a good unit test to add, hasn't been done though.
> @Quick-Flash - could you do it?  I cannot get my head around unit tests, they drive me nuts.

Unit testing C code is always a pain. I never wrapped my head around how BF does it unfortunately. One thing though that would help with unit testing is breaking up code into smaller functions that can be individually tested.

Reply:
@pichim: Maybe look at https://github.com/betaflight/betaflight/blob/master/src/test/unit/arming_prevention_unittest.cc / https://github.com/betaflight/betaflight/blob/master/src/test/unit/flight_failsafe_unittest.cc. Not the simplest test, but all necessary plumbing shall be obvious from it

@Quick-Flash : Unittests are actually in C++, using gtest framework. A lot of improvement is possible, but does work where you gent used to it.